Src/node_editor.py

```
@@ -168,32 +175,32 @@ def delink_callback(self, sender: int | str, app_data: int | str):
 
-        node_out: AbstractNode = dpg.get_item_user_data(dpg.get_item_parent(attr_outgoing))
-        node_in: AbstractNode = dpg.get_item_user_data(dpg.get_item_parent(attr_incoming))
+        node_out: AbstractNode = get_userdata(dpg.get_item_parent(attr_outgoing))
+        node_in: AbstractNode = get_userdata(dpg.get_item_parent(attr_incoming))
 
         node_out.outgoing[attr_outgoing].remove(attr_incoming)
         node_in.incoming[attr_incoming].remove(attr_outgoing)
 
-        dpg.set_item_user_data(attr_outgoing, node_out.outgoing[attr_outgoing])
-        dpg.set_item_user_data(attr_incoming, node_in.incoming[attr_incoming])
+        set_userdata(attr_outgoing, node_out.outgoing[attr_outgoing])
```

Ruff (0.15.20)

[error] Undefined name `set_userdata`

(F821)